extensions/mongodb/src/test/java/com/hazelcast/jet/mongodb/dataconnection/MongoDataConnectionMutualSslTest.java

```diff
@@ -0,0 +1,160 @@
+import com.mongodb.client.MongoClients;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.MongoDBContainer;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.util.ArrayList;
+
+import static com.hazelcast.jet.impl.util.ExceptionUtil.rethrow;
+import static com.hazelcast.jet.mongodb.AbstractMongoTest.TEST_MONGO_VERSION;
+import static com.hazelcast.jet.mongodb.impl.Mappers.defaultCodecRegistry;
+import static com.hazelcast.test.DockerTestUtil.assumeDockerEnabled;
+import static org.testcontainers.containers.BindMode.READ_WRITE;
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class, IgnoreInJenkinsOnWindows.class})
+public class MongoDataConnectionMutualSslTest extends SimpleTestInClusterSupport {
+
+    private static final String DATABASE = "MongoDataConnectionSslTest";
+    private static final String TEST_COLLECTION = "testCollection";
+    private static final Logger LOGGER = LoggerFactory.getLogger(MongoDataConnectionMutualSslTest.class);
+    private static final MongoDBContainer mongoContainer = new MyMongoContainer()
+            .withEnv("MONGO_INITDB_DATABASE", DATABASE)
+            .withClasspathResourceMapping("certs/localhost.pem", "/data/ssl/key.pem", READ_WRITE)
+            .withClasspathResourceMapping("certs/ca-cert.pem", "/data/ssl/ca.pem", READ_WRITE)
+            .withClasspathResourceMapping("tlsMongoMutual.yaml", "/etc/mongo/mongod.conf", READ_WRITE)
+            .withExposedPorts(27017)
+            .withLogConsumer(new Slf4jLogConsumer(LOGGER))
+            .withCommand("--config", "/etc/mongo/mongod.conf");
+    private static String connectionString;
+    private static String trustStoreLocation;
+    private static String keyStoreLocation;
+
+    @BeforeClass
+    public static void setUp() {
+        try (
+                InputStream resourceTS = MongoDataConnectionMutualSslTest.class.getResourceAsStream("/certs/ca.p12");
+                InputStream resourceKS = MongoDataConnectionMutualSslTest.class.getResourceAsStream("/certs/localhost.p12")
+        ) {
+            File tempFileTS = File.createTempFile("MongoDataConnectionSslTest", "jks");
+            Files.copy(resourceTS, tempFileTS.toPath(), StandardCopyOption.REPLACE_EXISTING);
+            File tempFileKS = File.createTempFile("MongoDataConnectionSslTest", "jks");
+            Files.copy(resourceKS, tempFileKS.toPath(), StandardCopyOption.REPLACE_EXISTING);
+
+            System.setProperty("https.protocols", "TLSv1,TLSv1.1,TLSv1.2,TLSv1.3");
+            System.setProperty("javax.net.debug", "ssl:handshake");
```

Comment:
The system properties reconfiguration shouldn't be needed.

Reply:
Right, removed